### ISSUE TITLE
Emscripten's port of SDL_Event causes a buffer overflow when member of a class

### ISSUE THREAD

**Issue by GameOverture** [emscripten-core/emscripten#13357]
When declaring a data member of `SDL_Event` in a class, and passing it into `SDL_PollEvent()` will stomp on 8 bytes of memory beyond its bounds.

Perhaps it's related, if you look at Emscripten's port of `typedef union SDL_Event` in SDL_events.h vs. SDL v2.0.12 it's missing:
```
    // This is necessary for ABI compatibility between Visual C++ and GCC
    //  Visual C++ will respect the push pack pragma and use 52 bytes for
    //  this structure, and GCC will use the alignment of the largest datatype
    //  within the union, which is 8 bytes.

    //   So... we'll add padding to force the size to be 56 bytes for both.
    
    Uint8 padding[56];
```

Removing the `SDL_event` data member from the class, and instead pushing a temp `SDL_event` on the stack before sending it into `SDL_PollEvent()` has resolved the memory stomping.

For testing sake, I tried also pushing some temp variables on the stack after the temp `SDL_event` and logging if those values were ever getting written to, and they weren't 🤷

**Comment by kripken** [comment]
Is this with SDL1 or 2?

**Comment by GameOverture** [comment]
SDL2

**Comment by kripken** [comment]
Thanks @GameOverture 

cc @Daft-Freak - is this a known issue perhaps?

**Comment by Daft-Freak** [comment]
Hmm, we're not modifying that header... and that padding _is_ there (https://github.com/emscripten-ports/SDL2/blob/2695e113f0878e49d7ac3f4953417f51d0ee6e0e/include/SDL_events.h#L594)

Though the old SDL 1 headers don't have it, might want to make sure you're getting the right headers. (Also compiling with `-s USE_SDL=2`?)

**Comment by GameOverture** [comment]
It looks like I wasn't including `-s USE_SDL=2` on the **COMPILE_FLAGS** within CMake, and only on the **LINK_FLAGS** properties. I assume that resulted with including the SDL1 headers while linking against the SDL2 library.

After fixing that (`sizeof(SDL_Event)` now equals 56) I'm getting the same runtime crash I had when all this started:
`RuntimeError: memory access out of bounds`
I will continue investigating tomorrow.

**Comment by GameOverture** [comment]
I've discovered when I `#include <GL/glew.h>` it will try to use SDL1 headers within `emscripten\system\include`. It does so on line 36: `#include <SDL/SDL_opengl.h>`

Is there a proper way to use glew with SDL2? In the interim I'm trying to get glad to work as an alternative to avoid any conflicts.

**Comment by GameOverture** [comment]
I've worked around this not using glew and instead `#include <GLES3/gl3.h>` which I guess is the more proper way of porting desktop OpenGL code with Emscripten? I no longer have certain identifiers that glew was offering but maybe those don't actually exist with ES3->webGL mappings.

I've reworked my CMakeLists file to make sure `-s USE_SDL=2` was hitting every library dependency as well as some other things and that seems to have hopefully solved this whole issue. Thanks for the help!
